File: 2025-10-07 - Padroes de Projetos - Iterator/Exemplo_Python/EstanteDeLivros.py
from EstanteIterator import EstanteIterator

class EstanteDeLivros:
   def __init__(self):
      self.__livros = []

   def getLivros(self):
      return self.__livros

   def setLivros(self, livros):
      self.__livros = livros

   def adicionar(self, livro):
      self.__livros.append(livro)

   def __iter__(self):
      return EstanteIterator(self.__livros)

File: 2025-10-07 - Padroes de Projetos - Iterator/Exemplo_Java/EstanteDeLivros.java
import java.util.ArrayList;
import java.util.Iterator;
import java.util.List;

public class EstanteDeLivros implements Iterable<Livro> {
   private List<Livro> livros;

   public EstanteDeLivros() {
      this.livros = new ArrayList<>();
   }

   public List<Livro> getLivros() {
      return livros;
   }

   public void setLivros(List<Livro> livros) {
      this.livros = livros;
   }

   public void adicionar(Livro livro) {
      this.livros.add(livro);
   }

   @Override
   public Iterator<Livro> iterator() {
      return new EstanteIterator(this.livros);
   }
}
